File: frontend/src/services/documentService.js
import api from './api';

/**
 * 文档服务 - 处理文档相关的API调用
 */
const documentService = {
  /**
   * 上传文档
   * @param {File} file - PDF文件
   * @returns {Promise} 上传结果
   */
  uploadDocument: async (file) => {
    const formData = new FormData();
    formData.append('file', file);

    return api.post('/documents/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
  },

  /**
   * 获取所有文档列表
   * @returns {Promise} 文档列表
   */
  getAllDocuments: async () => {
    return api.get('/documents');
  },

  /**
   * 根据ID获取文档详情
   * @param {number} id - 文档ID
   * @returns {Promise} 文档详情
   */
  getDocumentById: async (id) => {
    return api.get(`/documents/${id}`);
  },

  /**
   * 删除文档
   * @param {number} id - 文档ID
   * @returns {Promise} 删除结果
   */
  deleteDocument: async (id) => {
    return api.delete(`/documents/${id}`);
  },

  /**
   * 获取文档统计信息
   * @returns {Promise} 统计信息
   */
  getDocumentStats: async () => {
    return api.get('/documents/stats');
  },
};

export default documentService;

File: backend/src/main/java/com/scb/ragdemo/service/DocumentService.java
package com.scb.ragdemo.service;

import com.scb.ragdemo.model.dto.response.DocumentResponse;
import com.scb.ragdemo.model.entity.Document;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * Document Service Interface
 * Handles document upload, processing, querying, and deletion operations
 */
public interface DocumentService {
    
    /**
     * Upload a document file
     * 
     * @param file Uploaded file
     * @return Document response DTO
     */
    DocumentResponse uploadDocument(MultipartFile file);
    
    /**
     * Asynchronously process a document
     * Extracts text content and splits it into chunks for storage
     * 
     * @param documentId Document ID
     */
    void processDocumentAsync(Long documentId);
    
    /**
     * Get document by ID
     * 
     * @param id Document ID
     * @return Document entity
     */
    Document getDocumentById(Long id);
    
    /**
     * Get all documents with pagination
     * 
     * @param pageable Pagination parameters
     * @return Paginated document response DTOs
     */
    Page<DocumentResponse> getAllDocuments(Pageable pageable);
    
    /**
     * Get documents by processing status
     * 
     * @param status Document status
     * @return List of document response DTOs
     */
    List<DocumentResponse> getDocumentsByStatus(String status);
    
    /**
     * Delete a document and its associated chunks
     * 
     * @param id Document ID
     */
    void deleteDocument(Long id);
    
    /**
     * Reprocess a failed document
     * Deletes associated chunks and reprocesses the document
     * 
     * @param id Document ID
     * @return Document response DTO
     */
    DocumentResponse reprocessDocument(Long id);
    
    /**
     * Get document statistics
     * 
     * @return Statistics map
     */
    java.util.Map<String, Object> getDocumentStatistics();
}
